pkg/nsx/services/subnet/subnet.go

```diff
@@ -459,3 +461,39 @@ func (service *SubnetService) UpdateSubnetSet(ns string, vpcSubnets []*model.Vpc
 	}
 	return nil
 }
+
+// GetSharedSubnetFromNSX returns NSX Subnet, error and retriable for Shared Subnet CR.
+// If it fails due to API calls, the retriable will be true.
+func (service *SubnetService) GetSharedSubnetFromNSX(subnet *v1alpha1.Subnet) (*model.VpcSubnet, error, bool) {
+	anno := subnet.GetAnnotations()
+	associatedResource, ok := anno[common.AnnotationAssociatedResource]
+	if !ok {
+		return nil, fmt.Errorf("no associated resource annotation in Shared Subnet %s/%s", subnet.Namespace, subnet.Name), false
+	}
+	// associatedResource has the format projectID:vpcID:subnetID or :vpcID:subnetID for default project
+	parts := strings.Split(associatedResource, ":")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("failed to parse associated resource annotation %s", associatedResource), false
+	}
+	if parts[0] == "" {
+		nc, err := service.vpcService.GetDefaultNetworkConfig()
```

Comment:
for ":vpcID:subnetID", it means the default NSX Project, not the namespace's Project.

Reply:
Updated to use NSX API call to determine the default project, thanks